src/routes/token.rs

```diff
@@ -0,0 +1,50 @@
+use crate::{error::Error, AppState};
+use actix_web::{post, web, HttpResponse};
+use serde::{Deserialize, Serialize};
+use tokio::sync::RwLock;
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct TokenResponse {
+    pub access_token: String,
+    pub token_type: String,
+    pub refresh_token: String,
+    pub id_token: String,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct TokenRequestBody {
+    pub grant_type: String,
+    pub code: String,
+    pub redirect_uri: String,
+}
+
+#[post("/api/v1/token")]
+async fn post_token_handler(
+    app_state: web::Data<RwLock<AppState>>,
+    body: web::Form<TokenRequestBody>,
+) -> Result<HttpResponse, Error> {
```

Comment:
This should be a form and not JSON, according to spec.

Reply:
what exactly? the body is a Form the return is JSON 🤔 maybe you're referring to outdated code?

Reply:
Oh yeah, could be, sorry. I am 100% sure I saw JSON in there when I reviewed.